jans-auth-server/server/src/main/java/io/jans/as/server/authorize/ws/rs/AuthorizeRestWebServiceImpl.java

```diff
@@ -782,16 +790,37 @@ private Response requestAuthorization(
     }
 
     @Nullable
-    private ResponseMode extractResponseMode(String request) {
-        final Jwt jwt = Jwt.parseSilently(request);
-        if (jwt == null) {
+    private JsonWebResponse parseRequestToJweJwt(String request) {
+        String[] parts = request.split("\\.");
+        try {
+            if (parts.length == 5) {
+                String encodedHeader = parts[0];
+                JwtHeader jwtHeader = new JwtHeader(encodedHeader);
+                String keyId = jwtHeader.getKeyId();
+                PrivateKey privateKey = null;
+                KeyEncryptionAlgorithm keyEncryptionAlgorithm = KeyEncryptionAlgorithm
+                        .fromName(jwtHeader.getClaimAsString(JwtHeaderName.ALGORITHM));
+                if (AlgorithmFamily.RSA.equals(keyEncryptionAlgorithm.getFamily())) {
+                    privateKey = cryptoProvider.getPrivateKey(keyId);
+                }
+                Jwe jwe = Jwe.parse(request, privateKey, null);
+                if (jwe == null) {
+                    return null;
+                }
+                return jwe;
+            }
+            final Jwt jwt = Jwt.parseSilently(request);
+            if (jwt == null) {
+                return null;
```

Comment:
You can remove this if and simply do `return Jwt.parseSilently(request);`, it is redundant, because if it's null we anyway return it.